File: contest/LeetCode_Weekly_Contest_204/Detect_Pattern_of_Length_M_Repeated_K_or_More_Times.js
/**
 * @param {number[]} arr
 * @param {number} m
 * @param {number} k
 * @return {boolean}
 */
var containsPattern = function(arr, m, k) {
    var len = m * k;
    var n = arr.length;
    for (var i = 0; i + len <= n; ++i) {
        var flag = true;
        for (var j = i; j < i+len; j += m) {
            for (var idx = 0; idx < m; idx++)
                if (arr[i+idx] != arr[j+idx]) {
                    flag = false;
                    break;
                }
            if (!flag) break;
        }
        if (flag) return true;
    }
    return false;
};

File: contest/LeetCode_Weekly_Contest_204/Detect_Pattern_of_Length_M_Repeated_K_or_More_Times.java
class Solution {
    public boolean containsPattern(int[] arr, int m, int k) {
        int len = m * k;
        int n = arr.length;
        for (int i = 0; i + len <= n; ++i) {
            boolean flag = true;
            for (int j = i; j < i+len; j += m) {
                for (int idx = 0; idx < m; idx++)
                    if (arr[i+idx] != arr[j+idx]) {
                        flag = false;
                        break;
                    }
                if (!flag) break;
            }
            if (flag) return true;
        }
        return false;
    }
}